backend/src/test/java/com/huellapositiva/api/ProposalControllerShould.java

```diff
@@ -89,6 +93,90 @@ void beforeEach() {
+    }
+
+    @Test
+    void return_409_when_inadequate_criteria_to_change_proposal_status_to_finished() throws Exception{
+        // GIVEN
+        JpaProposal publishedProp= testData.registerESALAndProposal(CANCELLED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        JwtResponseDto jwtResponseDto = loginAndGetJwtTokens(mvc, DEFAULT_ESAL_CONTACT_PERSON_EMAIL, DEFAULT_PASSWORD);
+
+
+        // WHEN + THEN
+        mvc.perform(post(FETCH_PROPOSAL_URI + publishedProp.getId() + "/status/toFinished")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtResponseDto.getAccessToken())
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(csrf())
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isConflict())
+                .andExpect(mvcResult -> assertThat(jpaProposalRepository.findByNaturalId(publishedProp.getId()).get().getStatus().getName()).isEqualTo("cancelled"));
+
+    }
+
+    @Test
+    void return_200_when_adequate_criteria_to_change_proposal_status_to_finished() throws Exception{
+        // GIVEN
+        JpaProposal publishedProp= testData.registerESALAndProposal(PUBLISHED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        JwtResponseDto jwtResponseDto = loginAndGetJwtTokens(mvc, DEFAULT_ESAL_CONTACT_PERSON_EMAIL, DEFAULT_PASSWORD);
+
+
+        // WHEN + THEN
+        mvc.perform(post(FETCH_PROPOSAL_URI + publishedProp.getId() + "/status/toFinished")
+                .header(HttpHeaders.AUTHORIZATION, "Bearer " + jwtResponseDto.getAccessToken())
+                .contentType(MediaType.APPLICATION_JSON)
+                .with(csrf())
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(mvcResult -> assertThat(jpaProposalRepository.findByNaturalId(publishedProp.getId()).get().getStatus().getName()).isEqualTo("finished"));
+
+    }
+
+    @Test
+    void trigger_bulk_status_change_to_finished() throws Exception {
+
+        // GIVEN
+        JpaProposal finishedProp= testData.registerESALAndProposal(FINISHED);
+        finishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        testData.resetData();
+
+        JpaProposal publishedProp= testData.registerESALAndProposal(PUBLISHED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2099"));
+
+        testData.resetData();
+
+        JpaProposal enrollClosedProp= testData.registerESALAndProposal(ENROLLMENT_CLOSED);
+        publishedProp.setClosingProposalDate(new SimpleDateFormat("dd-MM-yyyy").parse("20-12-2020"));
+
+        testData.resetData();
+
+        // WHEN
+        proposalService.changeExpiredProposalStatusToFinished();
```

Comment:
This line is not doing anything at the moment as there are no proposals in the database